Zend/zend_mmap.h

```diff
@@ -0,0 +1,44 @@
+/*
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+   | Authors: Max Kellermann <max.kellermann@ionos.com>                   |
+   +----------------------------------------------------------------------+
+*/
+
+#ifndef ZEND_MMAP_H
+#define ZEND_MMAP_H
+
+#include "zend_portability.h"
+
+#ifdef __linux__
+# include <sys/prctl.h>
+
+/* fallback definitions if our libc is older than the kernel */
+# ifndef PR_SET_VMA
+#  define PR_SET_VMA 0x53564d41
+# endif
+# ifndef PR_SET_VMA_ANON_NAME
+#  define PR_SET_VMA_ANON_NAME 0
+# endif
+#endif // __linux__
+
+/**
+ * Set a name for the specified memory area.
+ *
+ * This feature requires Linux 5.17.
+ */
+static zend_always_inline void zend_mmap_set_name(const void *start, size_t len, const char *name)
+{
+#ifdef __linux__
+	prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, (unsigned long)start, len, (unsigned long)name);
```

Comment:
Is it safe to call `prctl` with options unsupported by the kernel?

https://github.com/torvalds/linux/blob/7403e6d8263937dea206dd201fed1ceed190ca18/kernel/sys.c#L2615

Looks like the answer is yes and it'll just return `-1` but the [docs](https://man7.org/linux/man-pages/man2/prctl.2.html)  aren't very clear so I thought I'd ask.

Reply:
> Is it safe to call `prctl` with options unsupported by the kernel?


yes,, it will just return -1;

Reply:
Not sure it warrants a new file for that.

Reply:
I made this a new file to reduce header dependencies, to avoid adding a dependency to `sys/prctl.h` to almost all sources.
The PHP code base is currently pretty clumsy about header dependencies; you have catch-all headers like `zend_portability.h`. I don't like that.
My coding style is to have many small and domain-specific headers with only the bare minimum of header-including-other-header. This speeds up compile times and improves code correctness.
But ... since this is not my project, I'll accept any style you desire, and will amend this PR.

Reply:
I feel it had more it maybe it would worth it but that s personal opinion tough, other members might think differently. But what do you make about the other points I made below ?

Reply:
Same sentiment here